src/lit-html.ts

```diff
@@ -332,63 +331,51 @@ export class NodePart extends Part implements SinglePart {
   private _setIterable(value: any): void {
     // For an Iterable, we create a new InstancePart per item, then set its
     // value to the item. This is a little bit of overhead for every item in
-    // an Iterable, but it lets us recurse easily and update Arrays of
-    // TemplateResults that will be commonly returned from expressions like:
-    // array.map((i) => html`${i}`)
-
-    // We reuse this parts startNode as the first part's startNode, and this
-    // parts endNode as the last part's endNode.
-
-    let itemStart = this.startNode;
-    let itemEnd;
-    const values = value[Symbol.iterator]() as Iterator<any>;
-
-    const previousParts: NodePart[]|undefined = Array.isArray(this._previousValue) ?
-        this._previousValue : undefined;
-    let previousPartsIndex = 0;
-    const itemParts = [];
-    let current = values.next();
-    let next = values.next();
-
-    if (current.done) {
-      // Empty iterable, just clear
+    // an Iterable, but it lets us recurse easily and efficiently update Arrays
+    // of TemplateResults that will be commonly returned from expressions like:
+    // array.map((i) => html`${i}`), by reusing existing TemplateInstances.
+
+    if (!Array.isArray(this._previousValue)) {
       this.clear();
+      this._previousValue = [];
```

Comment:
Because I don't want to clear it if the previous values already is an array, since it will contain the NodeParts from the previous render. Adding a comment.